test/types/records/generic/issue-22011-resolve-warning.chpl

```diff
@@ -0,0 +1,11 @@
+record generic_record {
+    var x;
+}
+
+record wrapper_record {
+    type t;
+}
+
+record R {
+    var field: wrapper_record(generic_record(?));
+}
```

Comment:
Since Chapel doesn't do much resolving of things until they're used, I think this test would be stronger if it declared some instances of the records in question.  However, my attempts to do that with code like the following failed, complaining again about partial instantiations on line 10:

```chapel
var gr = new generic_record(x=10);
writeln(gr, ": ", gr.type:string);
var r = new R(field = new wrapper_record(gr.type));
writeln(r, ": ", r.type:string);
```

Can you double-check me?

Reply:
This maybe feels like a place where the warning is incorrectly firing?

I used the following

```chapel
var gr = new generic_record(x=10);
writeln(gr, ": ", gr.type:string);
var wrap = new wrapper_record(gr.type);
writeln(wrap, ": ", wrap.type:string);
var r = new R(field = wrap);
writeln(r, ": ", r.type:string);
```

When I compile and run this, I get the same warning you did. But there is no place for me to add `(?)` that will resolve this warning. I get two warnings about `partial instantiation without '?' argument`. And if I remove `(?)` from `generic record` on line 10, I get three warnings. 


I also tried compiling with `--log` and `--print-id-on-error`. This reports ids that don't appear in the log. If I run the same test of removing `(?)` to get three warnings, I get a different behavior. Now 2 of the warnings have ids that exist for only `buildDefaultFunctions` and `createTaskFunctions`, and the other warning has an id that exists from parsing until `resolve`.

I _think_ the extra warnings are coming from the compiler generated default actual function and init= copy initializer. I semi-confirmed this by adding an explicit `init=`, which makes one of the 2 warnings go away.

```chapel
    proc init=(field: wrapper_record(generic_record(?))) {
        this.field = field;
    }
```

Reply:
That's a good guess.  So maybe all we need to do to squash the extra warnings is to have them not fire for compiler-generated code?  This is actually very similar to some issues I was hitting while taking a look at Michael's https://github.com/chapel-lang/chapel/pull/24021 a week or two ago.  Tagging in case he has insights or different thoughts here.

Reply:
@jabraham17 : I put together a quick and sloppy PR to try and squash these additional warnings, which appears to be working.  I'm going to paratest it to see whether it affects other existing tests (and whether that's positive or negative).

Reply:
Opened a PR in https://github.com/chapel-lang/chapel/pull/24339

Reply:
Since that PR should squash these extra warnings, I will add the usage code of these records to this PR and then wait for your PR to merge

Reply:
I did rebase and retest this after #24339, everything looks good!